src/main/scala/mesosphere/marathon/core/instance/update/InstanceUpdater.scala

```diff
@@ -121,4 +121,18 @@ object InstanceUpdater extends StrictLogging {
   private[marathon] def revert(instance: Instance): InstanceUpdateEffect = {
     InstanceUpdateEffect.Update(instance, oldState = None, events = Nil)
   }
+
+  private[marathon] def changeGoal(instance: Instance, op: InstanceUpdateOperation.ChangeGoal, now: Timestamp): InstanceUpdateEffect = {
+    val updatedInstance = instance.copy(state = instance.state.copy(goal = op.goal))
```

Comment:
This may be another issue but do we want to prevent from going from `Goal.Decommissioned` to `Goal.Running`? I think it's an illegal operation and would make the updates commutative.

Reply:
I'll follow up with another PR

Reply:
@jeschkies yes - thanks for highlighting that. I've been thinking generally about the same thing.